vm/src/obj/objlist.rs

```diff
@@ -170,6 +170,33 @@ fn list_index(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     Err(vm.new_value_error(format!("'{}' is not in list", needle_str)))
 }
 
+fn list_insert(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    trace!("list.insert called with: {:?}", args);
+    arg_check!(
+        vm,
+        args,
+        required = [
+            (list, Some(vm.ctx.list_type())),
+            (insert_position, Some(vm.ctx.int_type())),
+            (element, None)
+        ]
+    );
+    if objint::get_value(insert_position) > std::usize::MAX.into() {
+        return Err(
+            vm.new_overflow_error("Python int too large to convert to Rust usize".to_string())
+        );
+    }
+    let mut vec = get_mut_elements(list);
+    let position = match objint::get_value(insert_position) {
```

Comment:
It must be possible to implement this logic in a simpler manner. For example, this index does not need to be BigInt class but can be `i64` for example.

Reply:
My suggestion:
```rust
let index = objint::get_value(insert_position).to_i64();  // convert first to i64
let position = if (index < 0) {
  vec.len() - index
} else {
  index
};

// Bound checking:
if (position > vec.len()) {
  return Err("Index to big!");
} else if (position < 0) {
  return Err("Index to small");
}
let index = index.to_usize();
```

Reply:
I think I'll just convert it to `usize`, because that's the theoretical upper bound for the list size. I can also do a `if` to handle case when the index is larger then the usize.